### PR TITLE
Google Maps PR

### PR DESCRIPTION
**Pull request by vonschlichten**
**In this pull request I:**

- Implemented a google maps component using google maps react.
- Updated code to use state variables through destructuring 
- Updated code to pass state variables as props from parent to children components.
- Updated state to contain latitude/longitude 
- Implemented chevron font icon as react component, renders up/down onClick.
- Implemented dynamic rendering of roomID directly through URL bar.

### CODE REVIEW

File: client/dist/index.css [hunk 1]
```diff
@@ -71,3 +71,12 @@ html, body, #root {
   font-weight: normal;
   color: rgb(0,157,156)
```
Comment (lndncole) [inline client/dist/index.css:72]: line 72 - add semi-colon

File: client/src/components/App.jsx [hunk 2]
```diff
@@ -36,13 +42,18 @@ class App extends React.Component {
   };
```
Comment (lndncole) [inline client/src/components/App.jsx:42]: line 40 - remove white space

File: client/src/components/App.jsx [hunk 3]
```diff
@@ -36,13 +42,18 @@ class App extends React.Component {
...
 
   render() {
+    const { hostName, city, country, description, gettingAround, guidebookId, lat, lng } = this.state;
```
Comment (lndncole) [inline client/src/components/App.jsx:45]: not completely critical - but consider alphabetizing your restructured variables for easy lookup and cleaner style

File: client/src/components/App.jsx [hunk 4]
```diff
@@ -36,13 +42,18 @@ class App extends React.Component {
...
-        <p>{this.state.description}</p>
-        <p>{this.state.gettingAround}</p>
-        <Details guidebookId={this.state.guidebookId} />
+        <p>{hostName}'s place is located in {city}, {country}.</p>
+        <p>{description}</p>
+        <p>{gettingAround}</p>
+
+          <Details guidebookId={guidebookId} gettingAround={gettingAround} />
+        <div  className="NeighborHoodMap">
+          <NeighborHoodMap lat={lat} lng={lng} />
```
Comment (lndncole) [inline client/src/components/App.jsx:55]: The components you have on line 53 and 55 aren't nested so no need to indent them here - put them on the same line as the `<p>` tags. You could even remove the white space on line 52 unless there was a reason you wanted to keep it there.

File: client/src/components/App.jsx [hunk 5]
```diff
@@ -36,13 +42,18 @@ class App extends React.Component {
...
+        </div>
       </div>
     );
   }
```
Comment (lndncole) [inline client/src/components/App.jsx:59]: semi-colon after this curly brace - line 59

File: client/src/components/Details.jsx [hunk 6]
```diff
@@ -24,14 +24,14 @@ class Details extends React.Component {
     let readMore = null;
```
Comment (lndncole) [inline client/src/components/Details.jsx:24]: On line 17 (not shown in this comment view) - you have "event" listed as input in to the `handleReadMoreClick` function but you don't seem to use it. Consider removing 'event'.

File: client/src/components/Details.jsx [hunk 7]
```diff
@@ -24,14 +24,14 @@ class Details extends React.Component {
...
-      readMore = <ReadMore />
+      readMore = <ReadMore gettingAround={this.props.gettingAround} />
       icon = faChevronUp;
     }
     return (
       <div>
         <div onClick={this.handleReadMoreClick}>
           <span className="readMore">Read more about the neighborhood</span>
-          <FontAwesomeIcon icon={icon} />
+            <FontAwesomeIcon className="chevron" icon={icon} />
```
Comment (lndncole) [inline client/src/components/Details.jsx:34]: <FontAwesomeIcon (line 34) is indented one more space over than it needs to be. Bring it back one.

File: client/src/components/Details.jsx [hunk 8]
```diff
@@ -24,14 +24,14 @@ class Details extends React.Component {
...
         </div>
         {readMore}
       </div>
```
Comment (lndncole) [inline client/src/components/Details.jsx:37]: line 39 - add semi-colon

File: client/src/components/NeighborhoodMap.jsx [hunk 9]
```diff
@@ -0,0 +1,34 @@
...
+import Details from './Details';
+import { Map, GoogleApiWrapper } from 'google-maps-react';
+const googleMapApi = require('../../../googleMapConfig');
+
+class NeighborHoodMap extends React.Component {
+  constructor(props) {
+    super(props);
+
+  }
+
```
Comment (lndncole) [inline client/src/components/NeighborhoodMap.jsx:11]: remove white space

File: client/src/components/NeighborhoodMap.jsx [hunk 10]
```diff
@@ -0,0 +1,34 @@
...
+    const mapStyles = {
+      width: '50%',
+      height: '50%',
+    };
+
+    const { lat, lng } = this.props;
+
+    return (
+        <Map
+          google={this.props.google}
```
Comment (lndncole) [inline client/src/components/NeighborhoodMap.jsx:23]: you're using this.props here (line 23) when you're restructuring this.props.lat and lang on line 19. Just throw this.props.google in to that same destructured line (19) and just write 'google' on line 23 - remove this.props to stay consistent.

File: client/src/components/NeighborhoodMap.jsx [hunk 11]
```diff
@@ -0,0 +1,34 @@
...
+          zoom={8}
+          style={mapStyles}
+          initialCenter={{ lat: lat, lng: lng}}
+        />
+    );
+  }
```
Comment (lndncole) [inline client/src/components/NeighborhoodMap.jsx:29]: semi colon on line 29

File: client/src/components/ReadMore.jsx [hunk 12]
```diff
@@ -5,7 +5,7 @@ const ReadMore = (props) => (
     <h3>Getting Around</h3>
     <p>{props.gettingAround}</p>
     <h3>Guidebook</h3>
-    <p>For more information, checkout the guidebook.</p>
+    <p>For more information, checkout the {props.guidebook}.</p>
```
Comment (lndncole) [inline client/src/components/ReadMore.jsx:8]: you can destructure your props here if you want too - to remain consistent with your code in other places. You can do it directly within the function parameter on line 3 (ask me about that if you want to learn more about it) or you can add a return line and wrap your function in curly braces. lemme know if i can make that clearer if you want.

File: client/src/index.jsx [hunk 13]
```diff
@@ -2,4 +2,9 @@ import React from 'react';
 import ReactDOM from 'react-dom';
 import App from './components/App';
 
-ReactDOM.render(<App roomId={11111112} />, document.getElementById('root'));
+let listingId = window.location.href.split('/')[4];
+if (!listingId || listingId.length !== 8) {
+  listingId = '11111142';
+}
```
Comment (lndncole) [inline client/src/index.jsx:8]: maybe a semi-colon here for good measure (line 8)

**Review by lndncole — CHANGES_REQUESTED** [review]
Lookin good little Bill!

File: client/src/components/NeighborhoodMap.jsx (hunk 10 above)
Comment (vonschlichten) [reply client/src/components/NeighborhoodMap.jsx:23]: I'm afraid this might interfere with how the google map component functions.

**Pull request merged**